### PR TITLE
Add config to turn on globally power

### LINKED ISSUE #1823: Add a config option to turn on power features
**Is your feature request related to a problem? Please describe.**
When using scala-cli as the scala runner whenever you want to use an advanced option you need to add `--power`, which might not be the best UX experience when using a lot of advanced options

**Describe the solution you'd like**
Have a way to turn it on as a default:
`scala config --power`

 or 

`scala config --power true`

or 

`scala config power true`

**Describe alternatives you've considered**
Always add `--power`

**Additional context**
When merging scala-cli and scala into one functionality it will be required to get a smooth transition for the users of previous scala-cli versions

### PR DESCRIPTION
**Pull request by lwronski**
Fixes #1823 . Part of #1700

### CODE REVIEW

File: modules/cli/src/main/scala/scala/cli/ScalaCli.scala [hunk 1]
```diff
@@ -194,6 +196,12 @@ object ScalaCli {
       // Enable ANSI output in Windows terminal
       coursier.jniutils.WindowsAnsiTerminal.enableAnsiOutput()
 
+    for {
+      configDb   <- ConfigDb.open(Directories.directories.dbPath.toNIO)
```
Comment (Gedochao) [inline modules/cli/src/main/scala/scala/cli/ScalaCli.scala]: why not do it straight on `isSipScala` variable initialization?

Comment (lwronski) [reply modules/cli/src/main/scala/scala/cli/ScalaCli.scala:200]: I would be changed in another PR as part of #1824

Comment (lwronski) [reply modules/cli/src/main/scala/scala/cli/ScalaCli.scala]: Thanks, so I moved this logic to `isSipScala` variable initialization.

**Review by Gedochao — APPROVED** [review]
LGTM, commented on some NIT stuff

File: modules/cli/src/main/scala/scala/cli/commands/RestrictedCommandsParser.scala [hunk 2]
```diff
@@ -36,7 +36,8 @@ object RestrictedCommandsParser {
         case Right(Some(_, arg, _)) if !arg.isSupported =>
           Left((
             Error.UnrecognizedArgument(
-              s"`${args(index)}` option is not supported in `scala` command.\n  Please run it with `scala-cli` command or with `--power` flag."
+              s"`${args(index)}` option is not supported in `scala` command.\n  Please run it with `scala-cli` command or with `--power` flag" +
```
Comment (Gedochao) [inline modules/cli/src/main/scala/scala/cli/commands/RestrictedCommandsParser.scala]: NIT using triple `"` would improve readability

File: modules/build/src/main/scala/scala/build/preprocessing/DirectivesProcessor.scala [hunk 3]
```diff
@@ -35,7 +35,8 @@ object DirectivesProcessor {
     ) =
       if (!allowRestrictedFeatures && handler.isRestricted)
         val msg =
-          "This directive is not supported with 'scala' command. Please run it with `scala-cli` command or with `--power` flag."
+          "This directive is not supported with 'scala' command. Please run it with `scala-cli` command or with `--power` flag" +
```
Comment (Gedochao) [inline modules/build/src/main/scala/scala/build/preprocessing/DirectivesProcessor.scala]: NIT using triple `"` would improve readability

**Pull request merged**